### DISCUSSION TITLE
Failed to load resource: the server responded with a status of 400 (Bad Request)

### DISCUSSION THREAD

**Discussion by bluesiru** [kimytsc/covid-rest-vaccine-macro#35] [Q&A] [answered]
카카오로 잔여백신 예약 시도중이며 콘솔 탭에 제목과 같은 에러 코드가 뜨네요
또 화면상 예약시도위치 확인에 지도가 뜨지 않습니다. 시간은 1초당 한번 정도 계속 바뀌고 있구요
계속 뒀더니 403코드와 함께 예약시도에 실패한 내역도 하나 뜨네요 되는 것 같긴 한데 400코드가 뜨는게 계속 신경쓰입니다.
무엇이 문제인가요 ?

![캡처](https://user-images.githubusercontent.com/88699240/128806639-544fd02f-22f7-442d-a328-d359e3974024.PNG)

**Comment by kimytsc** [comment]
/api/v2/reservation 403 로그는 예약 시도 실패시 나타나는 로그입니다.

카카오는 예약 시도 실패시 403 상태코드로 응답을 주는데, 이 과정에서 무조건적으로 `Failed to load resource: the server responded with a status of 403 ()`이 로그에 노출되게 됩니다.

왼쪽 화면에서처럼 예약시도 내역에 실패 내역이 떠있으면 403으로 인한 데이터 갱신으로 정상입니다.

**Comment by kimytsc** [comment, marked as answer]
400의 경우, 지도를 못불러와서 생긴 로그인데, 혹시 디버깅을 위해 설정하신 map url이나 bounds를 알려주실 수 있을까요?

**Reply by bluesiru** [reply]
https://m.place.naver.com/rest/vaccine?vaccineFilter=unused&x=129.0754902&y=35.4344214&bounds=128.4163105%3B35.1999453%3B129.7346698%3B35.6682167

MAP URL 상기와 같이 진행중입니다.
그럼 말씀하신대로 지도를 불러오지 못해 생긴 로그인거라면, 지도만 안보일 뿐 예약에는 문제가 없을까요 ?
시간이 계속 갱신되고 예약시도에 실패하는 내역이 뜨는걸로 보아, 탐색 및 예약시도 자체는 정상적으로 진행되고 있는걸로 판단되긴 합니다.

**Reply by kimytsc** [reply]
map은 참고하기위해 임의로 만든것이라 에러나도 기능에 문제는 없습니다.

처음 주신 화면으로 보아 정상 동작하고 있습니다.

**Reply by kimytsc** [reply]
지도가 너무 크면 이미지가 노출되지 않는 이슈가 있어 추후 수정할 예정입니다.

다만, 해당 지도를 네이버에서 확인해보니 900여개의 리스트가 확인되는데 화면상의 카카오 리스트에서는 60여개만 나오고 있습니다. 아마도 카카오측에서 너무 큰 범위에 대해서는 중심을 기준으로 일정 범위 안에 나오는 것만 추려서 전달해 주는 것으로 추측됩니다.

지도 범위를 조금 축소시키고 중심을 잘 잡아 리스트를 한번 정리하시는게 좋을 것 같습니다.

**Reply by bluesiru** [reply]
답변 감사합니다 혹시 그럼 MAP URL을 축소하여 각기 다른 MAP URL을 따고 창을 2개로 분리하여 돌리는데에는 문제가 없을까요 ?

**Reply by kimytsc** [reply]
축소하여 2개 정도는 괜찮을거라 생각은 들지만, 그만큼 delay를 조금 높게 설정해서 차단의 위험성을 피하시는게 좋을 것 같습니다.
